### PR TITLE
Add CI run with HBWSpace

### PR DESCRIPTION
**Pull request by masterleinad**
Apparently, HBWSpace is not tested in our current CI and (therefore unsurprisingly) broken.

### CODE REVIEW

**Comment by masterleinad** [comment]
`CUDA-9.2-NVCC` failing in `default_exec.overlap_mdrange_policy` looks spurious.

File: core/src/Kokkos_Serial.hpp [hunk 1]
```diff
@@ -249,12 +249,22 @@ struct ZeroMemset<
...
 
+template <typename MemorySpace>
+struct MemorySpaceAccess<MemorySpace, Kokkos::Serial::scratch_memory_space> {
+  enum : bool { assignable = false };
+  enum : bool {
+    accessible =
+        MemorySpaceAccess<MemorySpace, Kokkos::Serial::memory_space>::accessible
+  };
+  enum : bool { deepcopy = false };
+};
```
Comment (Rombur) [inline core/src/Kokkos_Serial.hpp:260]: Why do you have this? The other host backends (OpenMP and Threads) don't need this.

Comment (masterleinad) [reply core/src/Kokkos_Serial.hpp]: Apparently, we don't need this anymore after #4896.

**Review by Rombur — APPROVED** [review]

**Comment by masterleinad** [comment]
We decoded to just remove `HBWSpace`.

**Pull request closed without merging**